app/src/main/java/com/github/polypoly/app/game/GameLobby.kt

```diff
@@ -1,30 +1,28 @@
 package com.github.polypoly.app.game
 
 import com.github.polypoly.app.game.user.User
-import kotlin.time.Duration
 
-class GameLobby(
-    val admin: User,
-    val gameMode: GameMode,
-    val minimumNumberOfPlayers: Int,
-    val maximumNumberOfPlayers: Int,
-    val roundDuration: Duration,
-    val gameMap: List<Zone>,
-    val initialPlayerBalance: Int,
-    val name: String,
-    val code: String,
+data class GameLobby(
+    val admin: User = User(),
+    val gameMode: GameMode = GameMode.LAST_STANDING,
+    val minimumNumberOfPlayers: Int = 2,
+    val maximumNumberOfPlayers: Int = Int.MAX_VALUE,
+    val roundDuration: Long = 0, //> unix timestamp encoding
+    val gameMap: List<Zone> = listOf(),
+    val initialPlayerBalance: Int = 0,
+    val name: String = "default-lobby-instance",
+    val code: String = "default-lobby-code",
     val private: Boolean = false
 ) {
 
-    private val currentUsersRegistered: ArrayList<User> = ArrayList()
-    val usersRegistered: List<User> get() = currentUsersRegistered.toList()
+    val usersRegistered: ArrayList<User> = ArrayList()
```

Comment:
Are you sure that changing this won't break the immutability?

Reply:
There is indeed a tradeoff that I made here: the initial version had better encapsulation as no outside class could directly add/remove users (they had to go through the specific functions), however I had to turn it that way because firebase expects to be able to directly set the users with an existing list of Users when fetching a User from the DB. With the previous version it didn't work :(

Reply:
Oh okay! Then it's a faire tradeoff :)